### ISSUE TITLE
Feature Request: Improve --devel --needed behavior

### ISSUE THREAD

**Issue by cryzed** [trizen/trizen#68]
Hey, back when I was using pacaur I usually updated my system completely using: `pacaur -Syu --devel --needed`. The behavior I observed when doing this was that pacaur updated all repositories of the development packages (`--devel`) and attempted to build the packages only if there were actually any new commits (`--needed`). This can of course only work well if trizen's cache directory is located somewhere semi-permanent, so that the state of the previously cloned repositories can actually be used for comparison.

However, if I use the same flags in trizen it will rebuild packages even if there weren't any new commits; the only thing it seems to skip is installing the built package if the same version is already installed on the system. Maybe there is a different combination of switches that I could use? If not, this would be nice to have.

**Comment by AladW** [comment]
https://github.com/trizen/trizen/issues/48

**Comment by quite** [comment]
As one of those leaving the pacaur-boat, I'd really like `--devel --needed` which only decides to upgrade a pkg built from a refreshed/checkedout repo if the pkgver() actually changes.

**Comment by parkerlreed** [comment]
Same here. Didn't realize source files didn't change and waited for a fairly long build, for it to just skip installing it at the end. Ideally a check at the beginning if the sources actually updated (pacaur behavior) would be great.

**Comment by t0m5k1** [comment]
Another here. 
Would love Trizen check the settings to see if cache dir has been changed from /tmp to a permanent path, then check to see when the last pull was made and compare that to the last commit of repo being checked. This way we would not need to wait for the package to be fully built before Trizen learns that version is already present.
Such a time waste

**Comment by Enverex** [comment]
This is indeed a very useful feature that aurman has (in that it compares the current pkgver against the installed one and only builds if it's changed). The only thing I'd request here is not to follow aurman's route entirely, in that aurman updates ALL repositories BEFORE starting any of the packages.

The problem with this is that if you want to update say, 1 program, it has to update 100 repositories before it even starts updating the first package. Each repo should be updated and checked before each specific package, not all in one giant go. The second issue is that if one of the package updates fails and you start again, it has to go through the whole repo updating process all-over again. So yeah, don't do it like that please.

**Comment by AladW** [comment]
Not sure why this thread is still open with people either power posting or posting nonsensical suggestions like supporting partial upgrades...

**Comment by opeeba** [comment]
Not sure why this thread is still being read by those that hate having fun.
I'm looking for trizen to cover use cases that are preventing me from
entirely jumping ship from pacaur. Why are you here Alad?

**Comment by AladW** [comment]
I'm here because I didn't bother to unfollow this thread. Anyway it's entirely up to the author if he wants to keep attracting this kind of spam.

**Comment by milkmiruku** [comment]
Partial upgrades are bad (though I reserve the right to randomly install things between system upgrades or ignore long package builds and hope things work ok).

I think the split is between caching source files and not. Trizen does it, but to /tmp by default so not persistently, and thus doesn't take advantage of that to work out what -git needs built or not.

Eh, it's hypocritical to call out someone out for something in a sentence that essentially does the same, though, lol, FWIW, independently of seeing the updates here, today I chose to go back to pacaur to get hassle free -git upgrades.

**Comment by Enverex** [comment]
Seems worth mentioning that "partial upgrades" doesn't really apply to AUR because they're not likely to be system packages or core libraries; Just because I've updated to the latest version of Barbie Seahorse Adventures, doesn't mean I also need to update to the latest version of the TI-86 Calculator emulator.
